core/Pos/Aeson/Core.hs

```diff
@@ -0,0 +1,62 @@
+-- | Aeson instances for core
+
+module Pos.Aeson.Core
+       (
+       ) where
+
+import           Universum
+
+import           Data.Aeson       (ToJSON (toJSON), FromJSON (..))
+import           Data.Aeson.TH    (defaultOptions, deriveToJSON, deriveJSON, deriveFromJSON)
+import           Data.Time.Units  (Millisecond, Microsecond)
+import           Serokell.Util.Base64  (JsonByteString (..))
+
+import           Pos.Aeson.Crypto ()
+import           Pos.Binary.Class (AsBinary (..))
+import           Pos.Core.Types   (ApplicationName (..), BlockCount (..), ChainDifficulty, CoinPortion,
+                                   Coin, EpochIndex, LocalSlotIndex, SharedSeed (..),
+                                   SlotCount (..), SlotId, BlockVersionData, SoftforkRule,
+                                   EpochIndex, Timestamp)
+import           Pos.Core.Fee     (TxFeePolicy, TxSizeLinear, Coeff)
+import           Pos.Core.Vss     (VssCertificate)
+
+instance FromJSON ByteString where
+    parseJSON v = getJsonByteString <$> parseJSON v
+
+instance ToJSON SharedSeed where
+    toJSON = toJSON . pretty
+
+instance FromJSON SharedSeed where
+    parseJSON v = SharedSeed <$> parseJSON v
+
+instance FromJSON (AsBinary w) where
+    parseJSON v = AsBinary <$> parseJSON v
+
+deriving instance ToJSON SlotCount
+
+-- NOTE: some of these types are used on frontend (PureScript).
```

Comment:
Oh, so we are sharing these instances with PureScript, it's sad :(

Reply:
@gromakovsky client-api (ie, purescript) will be soon (in a month) be removed and deprecated

Reply:
@akegalj does it apply to explorer too? Does explorer use these instances or its own wrappers?

Reply:
ah right. Not sure atm - I believe they are used. @sectore @ksaric do you know are these instances used by Explorer UI ?

Long term plan is also rewriting Explorer UI in javascript, but I don't think this will happen in 2017. (maybe around December)

Reply:
@akegalj AFAIK the Aeson instances for `explorer` are here - https://github.com/input-output-hk/cardano-sl/blob/master/explorer/src/Pos/Explorer/Aeson/ClientTypes.hs
The client types are here - https://github.com/input-output-hk/cardano-sl/blob/master/explorer/src/Pos/Explorer/Web/ClientTypes.hs

Reply:
@gromakovsky If it is problematic that they are shared with purescript, we can duplicate those instances for both Explorer and Middleware . Then they won't be shared.